packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -905,7 +905,7 @@ function mountImperativeHandle<T>(
 
   // TODO: If deps are provided, should we skip comparing the ref itself?
   const effectDeps =
-    deps !== null && deps !== undefined ? deps.concat([ref]) : [ref];
+    deps !== null && deps !== undefined ? deps.concat([ref]) : deps;
```

Comment:
```suggestion
    deps !== null && deps !== undefined ? deps.concat([ref]) : null;
```

Nit: Should only be null, not undefined. I wondered why this wasn't a type error but it's because `mountEffectImpl` and `updateEffectImpl` contain their own type check; we might inline those later.